lib/rspec/que/queue_up.rb

```diff
@@ -1,89 +1,155 @@
 require 'rspec/mocks/argument_list_matcher'
+require 'time'
 
 module RSpec
   module Que
     module Matchers
       class QueueUp
         def initialize(job_class = nil)
+          @matchers = [QueuedSomething.new]
+          @matchers << QueuedClass.new(job_class) if job_class
           @job_class = job_class
+
+          @stages = nil
         end
 
         def matches?(block)
           @before_jobs = enqueued_jobs.dup
           block.call
-          enqueued_something? && enqueued_correct_class? && with_correct_args?
+
+          @stages = []
+          @matched_jobs = new_jobs
+          @matchers.each do |matcher|
+            @stages << {matcher: matcher, candidates: @matched_jobs.dup}
+            @matched_jobs = @matched_jobs.delete_if {|job| !matcher.matches?(job) }
+          end
+          !@matched_jobs.empty?
         end
 
         def with(*args)
-          raise "Must specify the job class when specifying arguments" unless job_class
```

Comment:
It looked like this was an implementation-defined limitation rather than a deliberate gem API design decision, so I removed it. In support of this analysis: the line was not tested.